crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -63,105 +189,385 @@ impl TryFrom<&BankofamericaRouterData<&types::PaymentsAuthorizeRouterData>>
+                Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into())
+            }
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-// Auth Struct
-pub struct BankofamericaAuthType {
-    pub(super) api_key: Secret<String>,
-}
-
-impl TryFrom<&types::ConnectorAuthType> for BankofamericaAuthType {
-    type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
-        match auth_type {
-            types::ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: api_key.to_owned(),
-            }),
-            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
-        }
-    }
-}
-// PaymentsResponse
-//TODO: Append the remaining status flags
-#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
 pub enum BankofamericaPaymentStatus {
+    Authorized,
     Succeeded,
     Failed,
-    #[default]
-    Processing,
+    Voided,
+    Reversed,
+    Pending,
+    Declined,
+    AuthorizedPendingReview,
+    Transmitted,
 }
 
-impl From<BankofamericaPaymentStatus> for enums::AttemptStatus {
-    fn from(item: BankofamericaPaymentStatus) -> Self {
-        match item {
-            BankofamericaPaymentStatus::Succeeded => Self::Charged,
-            BankofamericaPaymentStatus::Failed => Self::Failure,
-            BankofamericaPaymentStatus::Processing => Self::Authorizing,
+impl ForeignFrom<(BankofamericaPaymentStatus, bool)> for enums::AttemptStatus {
+    fn foreign_from(item: (BankofamericaPaymentStatus, bool)) -> Self {
+        let status = match item.0 {
+            BankofamericaPaymentStatus::Authorized
+            | BankofamericaPaymentStatus::AuthorizedPendingReview => Self::Authorized,
+            BankofamericaPaymentStatus::Succeeded | BankofamericaPaymentStatus::Transmitted => {
+                Self::Charged
+            }
+            BankofamericaPaymentStatus::Voided | BankofamericaPaymentStatus::Reversed => {
+                Self::Voided
+            }
+            BankofamericaPaymentStatus::Failed | BankofamericaPaymentStatus::Declined => {
+                Self::Failure
+            }
+            BankofamericaPaymentStatus::Pending => Self::Pending,
+        };
+        if item.1 && status == Self::Authorized {
+            return enums::AttemptStatus::Pending;
         }
+        status
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[derive(Debug, Deserialize)]
 pub struct BankofamericaPaymentsResponse {
-    status: BankofamericaPaymentStatus,
     id: String,
+    status: BankofamericaPaymentStatus,
+    error_information: Option<BankofamericaErrorInformation>,
+    client_reference_information: Option<ClientReferenceInformation>,
```

Comment:
client_reference_information is optional incase of error_information and vice versa have 2 separate struct and handle BankofamericaPaymentsResponse as enum

Reply:
Done